<!DOCTYPE HTML>
<!--
	Landed by HTML5 UP
	html5up.net | @ajlkn
	Free for personal and commercial use under the CCA 3.0 license (html5up.net/license)
-->
<html>
	<head>
		<title>EvIL- Evolutionary Medicine</title>
		<meta charset="utf-8" />
		<meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no" />
		<link rel="stylesheet" href="assets/css/main.css" />
		<noscript><link rel="stylesheet" href="assets/css/noscript.css" /></noscript>
	</head>
	<body class="is-preload">
		<div id="page-wrapper">

			<!-- Header -->
			<header id="header">
				<h1 id="logo"><a href="index.html">Evolutionary Informatics Lab</a></h1>
				<nav id="nav">
					<ul>
						<li><a href="index.html">Home</a></li>
						<li><a href="research.html">Research</a>
							<ul>
								<li><a href="macro.html">Macroevolution</a></li>
								<li><a href="comp.html">Evolutionary Medicine</a></li>
								<li><a href="phylo.html">Phylogenomics</a></li>
								<li><a href="immuno.html">Immunogenetics</a></li>
							</ul>
						</li>
						<li><a href="current.html">Team</a>
						<ul>
							<li><a href="alex.html">Alex Dornburg</a></li>
							<li><a href="current.html">Current Team Members</a></li>
							<li><a href="alumni.html">Former Team Members</a></li>
						</ul>
					</li>
						<li><a href="publications.html">Publications</a></li>
						<li><a href="software.html">Software</a></li>
						<li><a href="media.html">Media</a></li>
						<li><a href="contact.html" class="button primary">Join Us</a></li>
					</ul>
				</nav>
			</header>

			<!-- Main -->
				<div id="main" class="wrapper style1">
					<div class="container">
						<header class="major">
							<h2>Evolutionary Medicine</h2>
							<p>Understanding our past helps us face our future</p>
						</header>
						<div class="row gtr-150">
							<div class="col-8 col-12-medium">

								<!-- Content -->
									<section id="content">
										<span class="image fit"><img src="images/evoMed/comp01.jpg" alt="" /></span>
										<h3>Making the connection</h3>
										<p>Ecological and evolutionary principles are central elements in organismal health. An understanding of ecology and evolution can greatly improve our ability to face some of the largest threats to human health including as antibiotic resistance, zoonotic transmission, or cancer. Our lab strives to leverage its phylogenomic expertise to aid in this endeavour </p>
										<p>We collaborate closely with other groups and have worked on topics that include the evolution of key oncogenes, the response of cancer to specific treatment, and most recently on the durability of the immune response following infection by SARS-CoV-2 </p>
										<h3>Using evolution to determine SARS-CoV-2 reinfection risks</h3>
										<span class="image left"><img src="images/evoMed/comp02.jpg" alt="" /></span>
										<p>Throughout the COVID-19 pandemic, there was much uncertainty about how long immunity lasts after someone who is unvaccinated is infected with SARS-CoV-2. Working together with <a href="#">Dr. Jeffrey Townsend</a> at the Yale School of Public Health and several other collaborators, we developed an innovative approach rooted in phylogenetic comparative methods to provide an answer: strong protection following natural infection is short-lived. </p>
										<p>We analyzed known reinfection and immunological data from the close viral relatives of SARS-CoV-2 that cause “common colds” — along with immunological data from SARS-CoV-1 and Middle East Respiratory Syndrome. Leveraging evolutionary principles, we were able to model the risk of COVID-19 reinfection over time.
										Reinfections can and have happened, even shortly after recovery. And they will become increasingly common as immunity wanes and new SARS-CoV-2 variants arise. The COVID-19 pandemic made it clear how interconnected we are, and these findings underscore the importance of vaccination to break transmission chains </p>

										<p>This model and findings were published long before reinfections were common or data on reinfection was available and have proven to be highly accurate<p>
										<h3>What about vaccination?</h3>
										<p>We have expanded our model to assess the risk of future infection by SARS-CoV-2 over time following vaccination from several vaccines. This provides a comparative basis for natural and vaccine mediated immunity as well as quantitative evidence that supports current booster schedules. </p>
										<h3>Lab initiatives</h3>
										<ul>
											<li>Expanding to other human infecting viruses</li>
											<li>Oncogene evolution</li>
											<li>Viral and bacterial surveys in marine environments</li>
											<li>Evolutionary techniques for pandemic preparedness</li>
										</ul>
									</section>

							</div>
							<div class="col-4 col-12-medium">

								<!-- Sidebar -->
									<section id="sidebar">
										<section>
											<h3>Projecting the seasonality of endemic COVID-19</h3>
											<p>It is clear that COVID-19 will become an endemic disease. However, when to expect surges of cases in different parts of the world is less clear. </p><p>On the one hand, SARS-CoV-2 could follow a pattern similar to other seasonal respiratory viruses such as the flu. On the other hand, it is possible the low levels of circulation occur year round. This would be similar to several other non-seasonal respiratory viruses</p><p> Using incidence data collected across the Northern Hemisphere from the closest evolutionary relatives of SARS-CoV-2, we projected what an endemic COVID-19 might look like and just published a preprint </p>
											<footer>
												<ul class="actions">
													<li><a href="/pubs/Townsend2022.pdf" class="button">Read the preprint</a></li>
												</ul>
											</footer>
										</section>
										<hr />
										<section>
											<a href="#" class="image fit"><img src="images/evoMed/comp03.jpg" alt="" /></a>
											<h3>Phylogeny and cancer</h3>
											<p> To effectively guide research in cancer biology it is important to place oncogenes into a phylogenetic context. We recently finished work on anaplastic lymphoma kinase (ALK), a gene present in species that span humans to fruitflies with a high degree of structural conservation.</p> <p> ALK has been identified as a potential target for therapeutic development due to its association with tumorigenesis in neuroblastoma, non small cell lung cancer, anaplastic large-cell lymphoma, breast and renal cell carcinomas. Our work demonstrated that the leukocyte tyrosinekinase (LTK) gene and ALK have switched functionality between ray-finned fishes and sarcopterygians (tetrapods, lungfish, and coelacanth) and that the interacting partners for these genes are not homologous between vertebrates and non-vertebrates. These findings have major implications for translational medicine </p> <p> In addition to this type of comparative genomics studies, we also recently worked several collaborators to use the phylogeny of a tumor in conjunction with patient history to reveal the selective response to different therapies. This type of approach has tremendous power for revealing general responses to therapies, and could pave the way fo getting ahead of cancer's evolutionary trajectory </p>
											<footer>
												<ul class="actions">
													<li><a href="/pubs/Fisk2022.pdf" class="button">Read the paper</a></li>
												</ul>
											</footer>
										</section>
									</section>

							</div>
						</div>
					</div>
				</div>

			<!-- Footer -->
			<footer id="footer">
				<ul class="icons">
					<li><a href="#" class="icon brands alt fa-github"><span class="label">GitHub</span></a></li>
					<li><a href="#" class="icon solid alt fa-envelope"><span class="label">Email</span></a></li>
				</ul>
				<ul class="copyright">
					<li>&copy; Evolutionary Informatics Lab. All rights reserved.</li><li>Design: <a href="http://html5up.net">HTML5 UP</a></li>
				</ul>
			</footer>

		</div>

		<!-- Scripts -->
			<script src="assets/js/jquery.min.js"></script>
			<script src="assets/js/jquery.scrolly.min.js"></script>
			<script src="assets/js/jquery.dropotron.min.js"></script>
			<script src="assets/js/jquery.scrollex.min.js"></script>
			<script src="assets/js/browser.min.js"></script>
			<script src="assets/js/breakpoints.min.js"></script>
			<script src="assets/js/util.js"></script>
			<script src="assets/js/main.js"></script>

	</body>
</html>
