<!DOCTYPE HTML>
<!--
	Landed by HTML5 UP
	html5up.net | @ajlkn
	Free for personal and commercial use under the CCA 3.0 license (html5up.net/license)
-->
<html>
	<head>
		<title>EvIL- Software</title>
		<meta charset="utf-8" />
		<meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no" />
		<link rel="stylesheet" href="assets/css/main.css" />
		<noscript><link rel="stylesheet" href="assets/css/noscript.css" /></noscript>
	</head>
	<body class="is-preload">
		<div id="page-wrapper">

			<!-- Header -->
			<header id="header">
				<h1 id="logo"><a href="index.html">Evolutionary Informatics Lab</a></h1>
				<nav id="nav">
					<ul>
						<li><a href="index.html">Home</a></li>
						<li><a href="research.html">Research</a>
							<ul>
								<li><a href="macro.html">Macroevolution</a></li>
								<li><a href="comp.html">Evolutionary Medicine</a></li>
								<li><a href="phylo.html">Phylogenomics</a></li>
								<li><a href="immuno.html">Immunogenetics</a></li>
							</ul>
						</li>
						<li><a href="current.html">Team</a>
						<ul>
							<li><a href="alex.html">Alex Dornburg</a></li>
							<li><a href="current.html">Current Team Members</a></li>
							<li><a href="alumni.html">Former Team Members</a></li>
						</ul>
					</li>
						<li><a href="publications.html">Publications</a></li>
						<li><a href="software.html">Software</a></li>
						<li><a href="media.html">Media</a></li>
						<li><a href="contact.html" class="button primary">Join Us</a></li>
					</ul>
				</nav>
			</header>

			<!-- Main -->
				<div id="main" class="wrapper style1">
					<div class="container">
						<header class="minor">
							<h2>PhyInformR</h2>
						</header>
						<!-- Content -->
							<section id="content">
								<div class="div1"> 
								<span class="image"><img src="images/PhyInformR/informrlogo.png" height="200" alt="" /></span>
								<h3>Overview</h3>
								<div class="custom_bold"><p> This part focuses on visualization. </p>
									<p>We anticipate this to be an area of rapid growth so please check back often for new ways to visualize these metrics.<br><br> Also, if you develop a new way to view these data please let us know and we will gladly post the code and link(s) to your manuscript(s)</p>
									<br>
									<br>
									
									
						<section id="main" class="EVwrapper style1">
							<div class="title">First steps</div>
							<div class="EVcontainer">
								<p>The quantitative framework of quartet internode calculations lends itself wonderfully to the development of new ways to visualize information in a given dataset.</p> 
								<p>This section highlights graphics from a few recent publications</p>
							
							Hwang et al. 2015 depicted QIRP across an entire tree by plotting the QIRP of a marker for each node simultaneously. This plot can be drawn by providing a tree, rate vector, and state space as in section 3. For this example, we will use a single tree from Near et al. 2014 along with site rates from the same study
							<pre style="text-align: left; white-space: pre-line;"><code>library(ape) 
								trees<-read.tree(system.file("extdata","polypterus_trees.phy",package="PhyInformR"))
								rate_vector<-as.matrix(rag1)
								bichir_tree <-trees[[1]]
								PlotTreeSI(bichir_tree,rate_vector,3) </code></pre>	
							<br>
							<span class="image"><img src="images/PhyInformR/informR_11.png" alt="" /></span>
							<br>
							Here the branch lengths (x axis, time) and the blue lines (y axis, QIRP) match up. 
							<br>We can see that a similar trend to the one at the end of the previous section: 
							<br>
							<br>
							<strong>Short internodes have lower QIRP and are more likely to be predicted to be either impacted by homoplasy or contain little information</strong>
							<br>
							<br>
							Here is another example based on fungal data illustrating this general point that uses this approach
							<br>
							<span class="image"><img src="images/PhyInformR/QIRPAcrossTree.png" alt="" /></span>
							<br>
							This approach to visualization can be quite handy when comparing markers, so leave this graphics window open for now
							QIRPAcrossTree.png
							</div>
				</section>
				<section id="main" class="EVwrapper style2">
						<div class="title">Visualizing Experimental Design</div>
						<div class="EVcontainer">	
							<br>
							Say we are interested choosing a new marker to sequence for bichirs. 
							<br>
							<br>
							We can compare the predicted utility of rag1 to candidate markers using this plotting method. In this example we will compare rag1 with the first locus from Prum et al. 2015
							We'll start by isolating the locus
							<pre style="text-align: left; white-space: pre-line;"><code>candidate.locus<-as.matrix(prumetalrates[1:1594])</code></pre>
							Now we can compare to the above plot as follows
							<pre style="text-align: left; white-space: pre-line;"><code>Plot.Another.TreeSI(bichir_tree, candidate.locus,3,col="red",type=3)</code></pre>
							<img class="image" src="images/PhyInformR/informR_12.png" alt="">
							<br>
							In the above plot, we have added QIRP to the previous plot on the Y axis with time still on the X axis. Blue lines correspond to the QIRP values of the tree internodes for rag1 while red lines correspond to our candidate locus. This visualization reveals that the candidate marker is predicted to be of higher utility for resolving every node in the bichir tree. This sort of visualization can be a great heuristic for choosing probe sets or primers for cost and time effective sequencing of markers
							<br>
							<br>
							This form of visualization also displays overall trends of markers over time, and can help disentangle sources of error in tree inference. However, the above method requires a fixed internode length. Prum et al. 2015 provided an alternative visualization that accommodates for uncertainty in internode length  
							<br>
							<br>
							Note that here we are using the files from Prum et al. (2015), which are <a href='https://zenodo.org/record/30269?ln=en#.VfJJ-GRViko'> available on Zenodo (DOI 10.5281)</a> though you could also substitute site rates from your own markers here
							<br>
							<br>
				</div>
				
			</section>
								
					
					
			<section id="main" class="EVwrapper style1">
				<div class="title">Heatmap</div>
					<div class="EVcontainer">
						<br>
						Download the above files and <a href='https://github.com/carolinafishes/PhyInformR'> open prumetal_heatmap_rosetta.r that is hosted on the github repository with phyinformR </a>
						<br>
						Navigate to the "Phylogenetic_Informativeness" part of the downloaded directory 
						<br>
						Change the setwd() function at the top of the rosetta file to the directory path, save, then source
						<pre style="text-align: left; white-space: pre-line;"><code>setwd("~/yourpath/Zenodo/Phylogenetic_Informativeness")
							source(url("https://raw.githubusercontent.com/carolinafishes/PhyInformR/master/inst/extdata/prumetalrates.txt"))</code></pre>
							This file translates the rates into locus specific rate matrices for you to explore. For this tutorial we will sort some loci by length to ask whether size of the locus is associated with increased information content. We are going to visualize a random subset of loci for the sake of keeping the tutorial manageable, though feel free to explore the full data further!
							<pre style="text-align: left; white-space: pre-line;"><code>target_loci<-c("L216","L182","L149","L213","L184","L223","L305","L8","L203","L95","L41","L295","L107","L163","L21","L2","L325","L80","L28") 
								for (i in 1:length(target_loci)){
									sizeInfo<-length(get(target_loci[i]))
									assign(paste0("le",target_loci[i]), sizeInfo)	
								}
								ll<-c(leL216,leL182,leL149,leL213,leL184,leL223,leL305,leL8,leL203,leL95,leL41,leL295,leL107,leL163,leL21,leL2,leL325,leL80,leL28)
								names(ll)<-target_loci
								rev(sort(ll))
							</code></pre>
							The guide tree from Prum et al. 2015 used relative rates (root height=1), so we are going to look at resolving Neoaves with a crown at .30. We begin by calculating the signal for different internode lengths. Note that the loci were already formatted as matrices in the Prum et al. 2015 script.
							<pre style="text-align: left; white-space: pre-line;"><code>#get the concatenated alignment
								space.maker(allrates,.30,3)->a1 
								#add another 19 loci sorted by size
								for (i in 1:length(target_loci)){
									tempspace<-space.maker(get(target_loci[i]),.3,3)
									j<-i+1
									assign(paste0("a",j), tempspace)
								}</code></pre>
								We now take this output and assemble our heatmaps.
								<pre style="text-align: left; white-space: pre-line;"><code>
									demo<-rbind(a1,a2,a3,a4,a5,a6,a7,a8,a9,a10,a11,a12,a13,a14,a15,a16,a17,a18,a19,a20)
									demo2<-as.matrix(demo)
									row.names(demo2)<-c("allrates","L213","leL182","leL203","leL149","leL2","leL295","leL325","leL107","leL41","leL216","leL28", "leL184","leL8","leL95","leL80","leL163","leL21","leL305","leL223")
									byThis <-.30/20
									lilts<-seq(byThis,0.30-0.0001,by=byThis)
									colnames(demo2)<-lilts
									heatmap.2(demo2, Colv=F,Rowv=F, scale='none') </code></pre>
									<br>
									<br>
									<img class="image" src="images/PhyInformR/informR_13.png" alt=""> 
									<br> 
									This heatmap shows us the probability of correct resolution for increasingly small internodes. This map can be generated for other loci and the internode distances can be zoomed in on using the space.maker.narrow function.
									<br>
									<br>
					</div>
				</div>
			</section>
			<section id="main" class="EVwrapper style2">
							<div class="title">References</div>
							<div class="EVcontainer">	
								Dornburg, A., M. Friedman, and T. J. Near. 2015. Phylogenetic analysis of molecular and morphological data highlights uncertainty in the relationships of fossil and living species of Elopomorpha (Actinopterygii: Teleostei). Molecular Phylogenetics and Evolution 89:205-218.
								<br>
								<br>
								Dornburg, A., A. J. Moore, J. M. Beaulieu, R. I. Eytan, and T. J. Near. 2014. The impact of shifts in marine biodiversity hotspots on patterns of range evolution: evidence from the Holocentridae (squirrelfishes and soldierfishes). Evolution 69:146-161. 
								<br>
								<br>
								Graybeal, A. “The Phylogenetic Utility of Cytochrome B: Lessons from Bufonid Frogs” Molecular phylogenetics and evolution 1993;: 256–269. 
								<br>
								<br>
								Harmon LJ, Weir JT, Brock CD, Glor RE, Challenger W. GEIGER: investigating evolutionary radiations. Bioinformatics. 2008;24:129–31.
								<br>
								<br>
								Hwang, J., Q. Zhao, Z. L. Yang, Z. Wang, and J. P. Townsend. 2015. Solving the ecological puzzle of mycorrhizal associations using data from annotated collections and environmental samples–an example of saddle fungi. Environmental microbiology reports 7:658-667.
								<br>
								<br>
								Lanfear, R., B. Calcott, S. Y. W. Ho, and S. Guindon. 2012. PartitionFinder: combined selection of partitioning schemes and substitution models for phylogenetic analyses. Molecular Biology and Evolution 29:1695-1701.
								<br>
								<br>
								Lanfear, R., B. Calcott, D. Kainer, C. Mayer, and A. Stamatakis. 2014. Selecting optimal partitioning schemes for phylogenomic datasets. BMC Evolutionary Biology 14:82.
								<br>
								<br>
								McCallum E, Weston S. Parallel R. “O’Reilly Media, Inc.”; 2011.
								<br>
								<br>
								Near, T. J., A. Dornburg, M. Tokita, D. Suzuki, M. C. Brandley, and M. Friedman. 2014. Boom and bust: ancient and recent diversification in bichirs (Polypteridae: Actinopterygii), a relictual lineage of ray-finned fishes. Evolution 68:1014-1026.
								<br>
								<br>
								Neuwirth,E.RColorBrewer: ColorBrewer Palettes. 2014. R package version 1.1-2. https://CRAN.R-project.org/package=RColorBrewer
								<br>
								<br>
								Paradis E, Claude J, Strimmer K. APE: Analyses of Phylogenetics and Evolution in R language. Bioinformatics. 2004;20:289–90.
								<br>
								<br>
								Posada, D., and K. A. Crandall. 1998. Modeltest: testing the model of DNA substitution. Bioinformatics 14:817-818.
								<br>
								<br>
								Prum, R. O., J. S. Berv, A. Dornburg, D. J. Field, J. P. Townsend, E. Moriarty Lemmon, and A. R. Lemmon. Accepted. A Comprehensive Phylogeny of Birds (Aves) using Targeted Next Generation DNA Sequencing. Nature.
								<br>
								<br>
								Revell LJ. phytools: an R package for phylogenetic comparative biology (and other things). Methods Ecol. Evol. 2011;3:217–23
								<br>
								<br>
								Su, Z., Z. Wang, F. Lopez-Giraldez, and J. P. Townsend. 2014. The impact of incorporating molecular evolutionary model into predictions of phylogenetic signal and noise. Phylogenetics, Phylogenomics, and Systematics 2:11.
								<br>
								<br>
								Su Z, Townsend JP. Utility of characters evolving at diverse rates of evolution to resolve quartet trees with unequal branch lengths: analytical predictions of long-branch effects. BMC Evol. Biol. 2015;15:86.
								<br>
								<br>
								Townsend, J. P. 2007. Profiling phylogenetic informativeness. Systematic Biology 56:222-231. 
								<br>
								<br>
								Townsend, J. P., Z. Su, and Y. I. Tekle. 2012. Phylogenetic signal and noise: predicting the power of a data set to resolve phylogeny. Systematic Biology 61:835-849.
								<br>
								<br>
								Warnes,G.R., Bolker, B., Bonebakker, L., Gentleman, R., Liaw, W.H.A., Lumley, T., Maechler, M., Magnusson, A., Moeller, S., Schwartz, B., Venables B., gplots: Various R Programming Tools for Plotting Data. 2016. R package version 3.0.1. https://CRAN.R-project.org/package=gplots.
								<br>
								<br>
								Wickham H. ggplot2: Elegant Graphics for Data Analysis. Springer; 2016.
								<br>
								<br>
								Yang, Z. Fair-balance paradox, star-tree paradox, and Bayesian phylogenetics. Molecular biology and evolution 2007; 24(8), pp.1639-1655.
					</div>
				</div>
			</section>
			<section id="main" class="EVwrapper style1">
							<div class="title">Go back?</div>
							<div class="EVcontainer">	
				
				<hr />
				To refresh on calculating signal and noise to generate your own visualizations
				<br>
				<br>
				<a href="phyinformrPart2.html" class="button neon">Back to signal and noise</a>
				<hr />
				<br>
				Need PI profiles? You can go back to the first section here 
				<br>
				<br>
				<a href="phyinformrPart1.html" class="button neon">Return to Beginning</a>


					

					</div>

				</div>
</section>

			


			<!-- Footer -->
			<footer id="footer">
				<ul class="icons">
					<li><a href="#" class="icon brands alt fa-github"><span class="label">GitHub</span></a></li>
					<li><a href="#" class="icon solid alt fa-envelope"><span class="label">Email</span></a></li>
				</ul>
				<ul class="copyright">
					<li>&copy; Evolutionary Informatics Lab. All rights reserved.</li><li>Design: <a href="http://html5up.net">HTML5 UP</a></li>
				</ul>
			</footer>

		</div>

		<!-- Scripts -->
			<script src="assets/js/jquery.min.js"></script>
			<script src="assets/js/jquery.scrolly.min.js"></script>
			<script src="assets/js/jquery.dropotron.min.js"></script>
			<script src="assets/js/jquery.scrollex.min.js"></script>
			<script src="assets/js/browser.min.js"></script>
			<script src="assets/js/breakpoints.min.js"></script>
			<script src="assets/js/util.js"></script>
			<script src="assets/js/main.js"></script>

	</body>
</html>
