<!DOCTYPE HTML>
<!--
	Landed by HTML5 UP
	html5up.net | @ajlkn
	Free for personal and commercial use under the CCA 3.0 license (html5up.net/license)
-->
<html>
	<head>
		<title>EvIL- Software</title>
		<meta charset="utf-8" />
		<meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no" />
		<link rel="stylesheet" href="assets/css/main.css" />
		<noscript><link rel="stylesheet" href="assets/css/noscript.css" /></noscript>
	</head>
	<body class="is-preload">
		<div id="page-wrapper">

			<!-- Header -->
			<header id="header">
				<h1 id="logo"><a href="index.html">Evolutionary Informatics Lab</a></h1>
				<nav id="nav">
					<ul>
						<li><a href="index.html">Home</a></li>
						<li><a href="research.html">Research</a>
							<ul>
								<li><a href="macro.html">Macroevolution</a></li>
								<li><a href="comp.html">Evolutionary Medicine</a></li>
								<li><a href="phylo.html">Phylogenomics</a></li>
								<li><a href="immuno.html">Immunogenetics</a></li>
							</ul>
						</li>
						<li><a href="current.html">Team</a>
						<ul>
							<li><a href="alex.html">Alex Dornburg</a></li>
							<li><a href="current.html">Current Team Members</a></li>
							<li><a href="alumni.html">Former Team Members</a></li>
						</ul>
					</li>
						<li><a href="publications.html">Publications</a></li>
						<li><a href="software.html">Software</a></li>
						<li><a href="media.html">Media</a></li>
						<li><a href="contact.html" class="button primary">Join Us</a></li>
					</ul>
				</nav>
			</header>

			<!-- Main -->
				<div id="main" class="wrapper style1">
					<div class="container">
						<header class="minor">
							<h2>PhyInformR</h2>
						</header>
						<!-- Content -->
							<section id="content">
								<div class="div1"> 
								<span class="image"><img src="images/PhyInformR/informrlogo.png" height="200" alt="" /></span>
								<h3>Overview</h3>
								<div class="custom_bold"><p> This section covers the basics of how to quantify signal versus noise for a given phylogenetic problem</p><p> These quantifications are based on Townsend et al. (2012), who introduced theory that takes into the account the interplay of site rates, time, and internode length between species divergences to assess the predicted probabilty of data contributing to accurate topological resolution </p><p>Use of this method requires three inputs: 
									<br>
									<br>
									Site rates, state space, and internode lengths
									<br>
									<br>
									Site rates are covered above. Now we'll introduce the internode lengths and state spaces and will continue using the avian tree and site rates from Prum et al. (2015) that are distributed with phyinformR. In case these are not in memory already use <pre style="text-align: left; white-space: pre-line;"><code> library(ape) 
										tree<-read.tree(system.file("extdata","Prumetal_timetree. phy",package="PhyInformR"))
										rr<-as.matrix(prumetalrates)</code></pre><br></div>
						<section id="main" class="EVwrapper style1">
							<div class="title">Signal or Noise</div>
							<div class="EVcontainer">
								<p>There are three quantities that phyloInformeR calculates with regard to a specified internode given a set of site rates: Quartet Internode Resolution Probability (QIRP, "Quirp"), Quartet Internode Homoplasy Probability (QIHP, "Quip"), and Quartet Internode Polytomy Probability (QIPP, "Quippy"). <br>Townsend et al. (2012) introduced two ways to calculate these quantities: An analytical approximation and a Monte Carlo based solution. Both approaches depend on site rates and two user defined internode lengths, T (time from present) and t (internode)<br>
							<img class="image" src="images/phyinformR_1.png" max-height="300" alt="">
							<br>
							The theory of Townsend et al. defines T and t based on a phylogenetic quartet with even branch lengths. Under this assumption, T is the time from the present to the ancestor of a taxon (red in the example above) and t, the focal internode length (grey in the example above). Later in this section we will discuss how to perform similar analyses allowing for uneven quartets 
							<br>
							<br>
							Lets start by calculating approximate solutions for QIRP, QIHP, and QIPP
							<br>
							<br>
							Using the above illustration, you should have an idea of what T (time) and t (internode distance) you will want to use for your data. 
							<br>
							<br>
							For state space, a binary morphological matrix could be assessed by setting the state space to 2 or amino acid (20 or ~5), or other types of data with differing numbers of characters. If you are using nucleotides, despite having a four character states, Simmons et al. have demonstrated the state space to be better modelled using 3 states, so we will go with that for the remainder of this guide. 
							<br>
							Here is the implementation using T= 100 million year (Ma) and t= 0.5 Ma
							<br>
							<pre style="text-align: left; white-space: pre-line;"><code>Approximator(100,0.5, rr, 3) </code></pre>	
							<br>
							Alternatively you can use the Monte Carlo simulation approach1. Since the simulation is time consuming dependent on the number of simulations and the number of sites, we recommend that you do this on a cluster or while you are doing something else. The output is automatically recorded to file. The input looks similar to the approximation, though now you specify a file name for the probabilities and an image file name. Note that the two files must have different names! 
							<br>
							You can also toggle the screen output on and off by setting image to either "TRUE" or "FALSE". The simulation can be run in parallel so 
							<br>
							Remember to set your cores appropriately using the registerDoParallel(cores=8) 
							<br>
							Here is the function using the same T and t and 5000 simulations on a subset of sites to save time
							<pre style="text-align: left; white-space: pre-line;"><code>rr2<-as.matrix(prumetalrates[1:20000])
							parallel.cluster.signal.noise(100,0.5,rr2,5000,3,filename="test",imagename="testimage",image="FALSE</code></pre>	
							<img class="img" src="images/PhyInformR/informR_7.png" height="300" alt="">
							<br>	
				</div>
				
			</section>
				<section id="main" class="EVwrapper style2">
						<div class="title">Advanced Quantification</div>
						<div class="EVcontainer">					
					
					<br>
					Calculations based on the equations of Townsend et al.(2012) make several assumptions 
					<br>
					<br>
					1) equal base frequency distributions in the alignment 
					<br>
					<br>
					2) equal branch lengths within the phylogenetic quartet 
					<br>
					<br>
					3) a Jukes-Cantor model of sequence evolution
					<br>
					<br>
					Su et al. (2014) provided extensions to the equations of Townsend et al.(2012) that enable any nucleotide substitution model or distribution of base frequencies and Su and Townsend (2015) provided the theoretical framework for relaxing the assumption of even branch lengths 
					<br>
					To use these extension, first specify your model settings based on a model selection program such as Modeltest, or Partitionfinder6. Model settings are defined as follows and require changing for different nucleotide substitution models, base frequencies, and branch lengths 
					<br>
					For example:
					<pre style="text-align: left; white-space: pre-line;"><code>a=1 
						b=1 
						c=1 
						d=1 
						e=1
						f=1 
						Pi_T=0.25
						Pi_C=0.25 
						Pi_A=0.25 
						Pi_G=0.25
						internode<-c(62.4937, 62.4937, 62.4937, 62.4937, 8.9939)
					</code></pre>
					Pi_T through Pi_G are the empirical base frequency parameters 
					<br>
					<br>
					<strong>IMPORTANT</strong> | these must sum up to one | Pi_T + Pi_C + Pi_A + Pi_G = 1 
					<br>
					<br>
					a through f are the relative rate parameters which are defined as follows
					<br> 
					<br> 
					rCT = rTC = a; rAT = rTA = b; rTG = rGT = c; rCA = rAC = d; rCG = rGC = e; rAG = rGA = f 
					<br>
					A table of relative rate parameter settings for different substitution models taken from Su et al. (2014) is attached for reference
					<img class="image" src="images/PhyInformR/informR_8.png" max-height="400" alt=""><br>
					Internode in the above code is an object containing the numerical values of branch lengths of T1, T2, T3, T4, and the internode length t0 for the four - taxon tree in question
					<br>
					<img class="image" src="images/phyinformR_1.png" max-height="300" alt="">
					<br> 
					IMPORTANT -T1 and T2 must belong to the same sister clade and T3 and T4 must belong to the other clade in the hypothesized topology of the four - taxon problem in question (see above)
					<br>
					<br>
					Quartet trees do not need to be rooted. All four subtending branches may vary
					<br>
					You can quantify QIHP, QIPP, and QIRP with
					<pre style="text-align: left; white-space: pre-line;"><code>allmodel.signal.noise (a,b,c,d,e,f,internode,Pi_T,Pi_C,Pi_A,Pi_G, rr)</code></pre>
					From left to right, the three values represent QIHP, QIPP, and QIRP
					</div>
				
			</section>
			<section id="main" class="EVwrapper style1">
				<div class="title">Posterior Distributions</div>
					<div class="EVcontainer">
						<br>
						Since branch length are rarely known with certainty, phyinformR can also be used to calculate QIRP, QIPP, and QIHP values across a distribution of trees such as those obtained from Bayesian analyses. 
						<br>
						For this example, we will read in use the sample distribution of bichir trees from a study by <a href='https://www.researchgate.net/publication/259528342_Boom_and_bust_ancient_and_recent_diversification_in_bichirs_Polypteridae_Actinopterygii_a_relictual_lineage_of_ray-finned_fishes?ev=prf_pub'> Near et al. 2015</a> that is provided with the release 
						<pre style="text-align: left; white-space: pre-line;"><code>library(ape) 
							trees<-read.tree(system.file("extdata","polypterus_trees.phy",package="PhyInformR"))
							rate_vector<-as.matrix(rag1)</code></pre>	
							First you will need to specify the quartet of interest. In the bichir dataset, we will look at the clade containing Polypterus congicus as this species was not placed with high support in the tree: <br>
							<img class="image" src="images/PhyInformR/phyinformR_bichir.png" alt="">
							<br>
							We define the quartet as follows
							<pre style="text-align: left; white-space: pre-line;"><code>quart<- c("Polypterus_congicus","Polypterus_bichir","Polypterus_ansorgii" ,"Polypterus_endlicheri" )</code></pre>
							The remaining objects should be familiar, please review the above and preceeding page if the variable names seem enigmatic. To compute over a distribution of trees, run the function. Note that here we are truncating the tree distribution for the purpose of example
							<pre style="text-align: left; white-space: pre-line;"><code>final<-su.bayes(a,b,c,d,e,f,Pi_T,Pi_C,Pi_A,Pi_G,rate_vector,quart,trees)</code></pre>
							This function returns a matrix of internodes and T values from the trees and their associated QIHP, QIPP, and QIRP values 
							<br>
							<br>
							su.bayes runs in parallel so remember to set your cores appropriately using registerDoParallel(cores=8) 
							<br>
							<br>
							su.bayes returns a matrix of internodes and T values from the trees and their associated QIHP, QIPP, and QIRP values. This matrix can be summarized using
							<pre style="text-align: left; white-space: pre-line;"><code>plotPosterior(final, plot="QIPs")   #or
								plotPosterior(final,plot="violin")</code></pre>
							<br>
							Setting plot="qips" returns a density plot of the quartet internode resolution/polytomy/homoplasy probabilities and the internode lengths
							<br>
							<img class="image" src="images/PhyInformR/informR_14.png" alt="">
							<br>
							Visualizing the density of calculations reveals a trend that is common in phylogenetic datasets. Both lack of information and increased probabilities of convergence misleading inference plague smaller internodes. In this plot we can see that the bulk of the posterior density is in the realm of low QIRP and high QIPP, so we can conclude that the lack of resolution of this clade by this locus is in part predicted to be driven by limited information content
							<br>
							<br>
							Plot = "violin" returns violin plots of the quartet internode resolution/polytomy/homoplasy probabilities and the internode lengths 
							<br>
							<img class="image" src="images/PhyInformR/informR_15.png" alt="">
							<br>
							Visualizing the quantiles and kernel density of calculations allows for a additional perspective of how topological and branch length uncertainty influence quantifications. 
							<br>
							In this case we can see from the box plot of quantiles that QIHP is generally low, but that QIPP is centered near 0.45 with the majority of trees leading to a calculation between about 0.5 and 0.35. The kernel density gives us additional perspective, showcasing somewhat inverted distributions between QIRP and QIPP, with the majority of QIRP values being lower. 
					</div>
				</div>
			</section>
			<section id="main" class="EVwrapper style2">
							<div class="title">Next Steps</div>
							<div class="EVcontainer">	
				
				<hr />
				Ok, now that we can calculate signal and noise, lets move on to some visualizations!
				<br>
				<br>
				<a href="phyinformrPart3.html" class="button neon">On to Visualization</a>
				<hr />
				<br>
				Need PI profiles? You can go back to the first section here 
				<br>
				<br>
				<a href="phyinformrPart1.html" class="button neon">Return to Beginning</a>


					

					</div>

				</div>
</section>

			


			<!-- Footer -->
			<footer id="footer">
				<ul class="icons">
					<li><a href="#" class="icon brands alt fa-github"><span class="label">GitHub</span></a></li>
					<li><a href="#" class="icon solid alt fa-envelope"><span class="label">Email</span></a></li>
				</ul>
				<ul class="copyright">
					<li>&copy; Evolutionary Informatics Lab. All rights reserved.</li><li>Design: <a href="http://html5up.net">HTML5 UP</a></li>
				</ul>
			</footer>

		</div>

		<!-- Scripts -->
			<script src="assets/js/jquery.min.js"></script>
			<script src="assets/js/jquery.scrolly.min.js"></script>
			<script src="assets/js/jquery.dropotron.min.js"></script>
			<script src="assets/js/jquery.scrollex.min.js"></script>
			<script src="assets/js/browser.min.js"></script>
			<script src="assets/js/breakpoints.min.js"></script>
			<script src="assets/js/util.js"></script>
			<script src="assets/js/main.js"></script>

	</body>
</html>
