<!DOCTYPE HTML>
<!--
	Landed by HTML5 UP
	html5up.net | @ajlkn
	Free for personal and commercial use under the CCA 3.0 license (html5up.net/license)
-->
<html>
	<head>
		<title>EvIL- Phylogenomics</title>
		<meta charset="utf-8" />
		<meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no" />
		<link rel="stylesheet" href="assets/css/main.css" />
		<noscript><link rel="stylesheet" href="assets/css/noscript.css" /></noscript>
	</head>
	<body class="is-preload">
		<div id="page-wrapper">

			<!-- Header -->
			<header id="header">
				<h1 id="logo"><a href="index.html">Evolutionary Informatics Lab</a></h1>
				<nav id="nav">
					<ul>
						<li><a href="index.html">Home</a></li>
						<li><a href="research.html">Research</a>
							<ul>
								<li><a href="macro.html">Macroevolution</a></li>
								<li><a href="comp.html">Evolutionary Medicine</a></li>
								<li><a href="phylo.html">Phylogenomics</a></li>
								<li><a href="immuno.html">Immunogenetics</a></li>
							</ul>
						</li>
						<li><a href="current.html">Team</a>
						<ul>
							<li><a href="alex.html">Alex Dornburg</a></li>
							<li><a href="current.html">Current Team Members</a></li>
							<li><a href="alumni.html">Former Team Members</a></li>
						</ul>
					</li>
						<li><a href="publications.html">Publications</a></li>
						<li><a href="software.html">Software</a></li>
						<li><a href="media.html">Media</a></li>
						<li><a href="contact.html" class="button primary">Join Us</a></li>
					</ul>
				</nav>
			</header>

			<!-- Main -->
				<div id="main" class="wrapper style1">
					<div class="container">
						<header class="major">
							<h2>Phylogenomics</h2>
							<p>Darwin's vision of a Tree of Life comes into view</p>
						</header>
						<div class="row gtr-150">
							<div class="col-8 col-12-medium">

								<!-- Content -->
									<section id="content">
										<a class="image fit"><img src="images/Phylo/Hiding2.jpg" alt="" /></a>
										<h3>Resolving the Tree of Life</h3>
										<p>Phylogenetic trees are an integral component of comparative analyses that span topics ranging from the transmission of SARS-CoV-2 to human cancer biology to the factors that have shaped patterns of uneven species richness across the planet. Understanding the phylogeny of major groups of organisms has attracted the attention of evolutionary biologists since Charles Darwin drafted his famous “I think” diagram in 1837 (Barrett et al. 1987). However, it took more than 100 years for Darwin’s abstract concept of the tree of life to develop into an explicit, data-driven approach to phylogeny reconstruction.</p>
										<p>The early twenty-first century represents a phylogenetic renaissance. Over the past 20 years, hundreds of researchers have contributed resolved molecular phylogenies at varying time scales across the diversity of life. We continue to this effort to resolve all branches of the Tree of Life thereby provide a comparative framework for biologists. However, there are numerous pitfalls along the way, and we also develop computational means to avoid these</p>
										<a class="image fit"><img src="images/Phylo/rattlesnake2.jpg" alt="" /></a>
										<h3>Avoiding Pitfalls</h3>
										<p>We employ quantitative measures from phylogenetic informativeness (PI) theory to provide a predictive framework with which to assess the influence of noise on phylogenetic datasets. The aims of this work are two-fold</p> <p>First, by creating a predictive framework for phylogenomic studies, markers or types of sequence data can be chosen objectively, saving investigator time and money by enabling cost-effective research design. Second, by scrutinizing existing datasets, sources of conflicting information can be revealed, leading to more stable inferences </p><p>These approaches are incorporated into computational pipelines used in most of our research efforts, and are available in the R package <a href="phyinformrMain.html">PhyInformR.</a> We continue making these methods approachable and site-rates for our anlyses can now be estimated in IQTree</p> 
										<a class="image fit"><img src="images/Phylo/mainSkull2.jpg" alt="" /></a>
										<h3> On the birth and death of genes </h3>
										<p>The decay of phylogenetic information and ability to predict where to expect sequence convergence is directly linked to the ability to detect sequence homologs. We are currently expanding theory aimed at assessing loss of phylogenetic information to the problem of ortholog detection. This addresses a major challenge at deep evolutionary time scales or for rapidly evolving loci. Our goal is to provide an immensely powerful framework for disentangling functional orthologs from genetic orthologs and assessing the roles of adaptation and gene birth/death events in the maintenance of genetic diversity </p> 

										<h3> Current Research Projects</h3>
										<ul>
											<li>Continue to resolve the ray-finned fish Tree of Life</li>
											<li>The role of compositional bias and taxon sampling</li>
											<li>Brachiobdellidan phylogenetics</li>
											<li>Ortholog detection and gene family phylogeny</li>
											<li>Experimental design for massive viral sequence datasets</li>
										</ul>
									</section>

							</div>
							<div class="col-4 col-12-medium">

								<!-- Sidebar -->
									<section id="sidebar">
										<section>
											<h3>Dataset Assembly and Missing Data</h3>
											<p>The amount of freely available sequence data has become staggering and has tremendous promise for science. However, how to effectively harness this resource is not trivial. Large sequence datasets can be messy, and we often need to aggregate data from multiple sources. Moreover, high levels of missing data can impact our analyses. Extracting the data we need from within this tangled sea of information requires bioinformatic tools </p> 
											<a href="#" class="image fit"><img src="images/Phylo/Maine.jpg" alt="" /></a>
											<p> To aid in this effort, we continue to develop <a href="">TOAST</a>, an R package that enables extraction of thousands of single copy orthologs for any taxonomic group using any combination of online or local data while also tracking missing data patterns. Building upon functions for assessing thresholds of acceptable missing data allows for custom data set creation, that work in concert with numerous utility functions that produce outputs directly for phylogenetic software such as BEAST or IQTREE. Our goal for TOAST is to provide readily accessible tools that can harvest any set of loci for any set of organisms with only a minimal knowledge of R required by end-users. This goal is in line with with our vision of empowering anyone to utilize public genomic scale datasets in their research</p>
											<footer>
												<ul class="actions">
													<li><a href="#" class="button">See the tutorial</a></li>
												</ul>
												
											</footer>
										</section>
										<hr />
										<section>
											<a href="#" class="image fit"><img src="images/pic07.jpg" alt="" /></a>
											<h3>Amet Lorem Tempus</h3>
											<p>Sed tristique purus vitae volutpat commodo suscipit amet sed nibh. Proin a ullamcorper sed blandit. Sed tristique purus vitae volutpat commodo suscipit ullamcorper sed blandit lorem ipsum dolore.</p>
											<footer>
												<ul class="actions">
													<li><a href="#" class="button">Learn More</a></li>
												</ul>
											</footer>
										</section>
									</section>

							</div>
						</div>
					</div>
				</div>

			<!-- Footer -->
			<footer id="footer">
				<ul class="icons">
					<li><a href="#" class="icon brands alt fa-github"><span class="label">GitHub</span></a></li>
					<li><a href="#" class="icon solid alt fa-envelope"><span class="label">Email</span></a></li>
				</ul>
				<ul class="copyright">
					<li>&copy; Evolutionary Informatics Lab. All rights reserved.</li><li>Design: <a href="http://html5up.net">HTML5 UP</a></li>
				</ul>
			</footer>

		</div>

		<!-- Scripts -->
			<script src="assets/js/jquery.min.js"></script>
			<script src="assets/js/jquery.scrolly.min.js"></script>
			<script src="assets/js/jquery.dropotron.min.js"></script>
			<script src="assets/js/jquery.scrollex.min.js"></script>
			<script src="assets/js/browser.min.js"></script>
			<script src="assets/js/breakpoints.min.js"></script>
			<script src="assets/js/util.js"></script>
			<script src="assets/js/main.js"></script>

	</body>
</html>
