<!DOCTYPE HTML>
<!--
	Landed by HTML5 UP
	html5up.net | @ajlkn
	Free for personal and commercial use under the CCA 3.0 license (html5up.net/license)
-->
<html>
	<head>
		<title>EvIL- Software</title>
		<meta charset="utf-8" />
		<meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no" />
		<link rel="stylesheet" href="assets/css/main.css" />
		<noscript><link rel="stylesheet" href="assets/css/noscript.css" /></noscript>
	</head>
	<body class="is-preload">
		<div id="page-wrapper">

			<!-- Header -->
			<header id="header">
				<h1 id="logo"><a href="index.html">Evolutionary Informatics Lab</a></h1>
				<nav id="nav">
					<ul>
						<li><a href="index.html">Home</a></li>
						<li><a href="research.html">Research</a>
							<ul>
								<li><a href="macro.html">Macroevolution</a></li>
								<li><a href="comp.html">Computational Epidemiology</a></li>
								<li><a href="phylo.html">Phylogenomics</a></li>
								<li><a href="immuno.html">Immunogenetics</a></li>
							</ul>
						</li>
						<li><a href="current.html">Team</a>
						<ul>
							<li><a href="alex.html">Alex Dornburg</a></li>
							<li><a href="current.html">Current Team Members</a></li>
							<li><a href="alumni.html">Former Team Members</a></li>
						</ul>
					</li>
						<li><a href="publications.html">Publications</a></li>
						<li><a href="software.html">Software</a></li>
						<li><a href="media.html">Media</a></li>
						<li><a href="contact.html" class="button primary">Join Us</a></li>
					</ul>
				</nav>
			</header>

			<!-- Main -->
				<div id="main" class="wrapper style1">
					<div class="container">
						<header class="major">
							<h2>Software</h2>
							<p>As scientists we constantly need to modify or develop software in order to better analyze our data.
								As our team develops new tools or tweaks existing programs to do useful things, we will be updating
								this section to contribute to the growing body of bioinformatic software.</p>
						</header>

						<!-- Content -->
							<section id="content">
								<h3>phyinformR: Phylogenetic experimental design and data exploration in R</h3>
								<p><span class="image left"><img src="images/PhyinformRCover.png" alt="" /></span> PhyInformR is an open source software package that facilitates rapid calculation of phylogenetic informativeness and several novel related assessments of the phylogenetic utility of any given dataset or dataset partition. </p><p>By facilitating computation on user hardware, PhyInformR increases the potential power users can apply toward screening datasets for phylogenetic information content by orders of magnitude. </p><p>PhyInformR also features expanded functionality compared to current resources, providing new tools for data visualization and routines optimized for rapid statistical calculations using parallel processing.</p>
								<div class="col-6 col-12-xsmall">
										<ul class="actions fit small">
											<li><a href="#" class="button primary fit small">PhyinformR Walkthrough</a></li>
										</ul>
									</div>
								<p></p>
								<br></br>
								<h3>TOAST: Transcriptome Ortholog Alignment Tools </h3>
								<p><span class="image left"><img src="images/TOAST_Manual_cover.png" height="250" alt="" /></span> TOAST is an open source R package designed to help you effectively harness public data, visualize missing data patterns, and assemble multiple sequence alignments.TOAST can be downloaded here and requires R. </p><p>TOAST is continually updated to include utility functions that make data parsing easier and circumvent the use of multiple software steps in informatic pipelines. TOAST can generate output that can be fed directly into IQTREE and has numerous functions for analyzing missing data and subsetting massive sequence datasets</p>
								<div class="col-6 col-12-xsmall">
										<ul class="actions fit small">
											<li><a href="#" class="button primary fit small">TOAST Walkthrough</a></li>
										</ul>
									</div>
								<br></br>
								<h3>Additional Scripts </h3>
								<p><span class="image left"><img src="images/toadfish3.png" height="250" alt="" /></span> Here we provide an overview of custom scripts that were created in the course of our projects  with detailed tutorials and additional content</p><p>This archive will be updated as new work is published and provide links to the corresponding data files allowing you to replicate our results while learning new techniques </p>
								<p> Please check back periodically for new updates!</p>
								<div class="col-6 col-12-xsmall">
										<ul class="actions fit small">
											<li><a href="#" class="button primary fit small">See Archive</a></li>
										</ul>
									</div>
								<p></p>
								<br></br>


								
								

							</section>

					</div>
				</div>

			<!-- Footer -->
			<footer id="footer">
				<ul class="icons">
					<li><a href="#" class="icon brands alt fa-github"><span class="label">GitHub</span></a></li>
					<li><a href="#" class="icon solid alt fa-envelope"><span class="label">Email</span></a></li>
				</ul>
				<ul class="copyright">
					<li>&copy; Evolutionary Informatics Lab. All rights reserved.</li><li>Design: <a href="http://html5up.net">HTML5 UP</a></li>
				</ul>
			</footer>

		</div>

		<!-- Scripts -->
			<script src="assets/js/jquery.min.js"></script>
			<script src="assets/js/jquery.scrolly.min.js"></script>
			<script src="assets/js/jquery.dropotron.min.js"></script>
			<script src="assets/js/jquery.scrollex.min.js"></script>
			<script src="assets/js/browser.min.js"></script>
			<script src="assets/js/breakpoints.min.js"></script>
			<script src="assets/js/util.js"></script>
			<script src="assets/js/main.js"></script>

	</body>
</html>
