<!DOCTYPE html>
<!--
	Landed by HTML5 UP
	html5up.net | @ajlkn
	Free for personal and commercial use under the CCA 3.0 license (html5up.net/license)
-->
<html>
  <head>
    <title>EvIL- Research</title>
    <meta charset="utf-8" />
    <meta
      name="viewport"
      content="width=device-width, initial-scale=1, user-scalable=no"
    />
    <link rel="stylesheet" href="assets/css/main.css" />
    <noscript
      ><link rel="stylesheet" href="assets/css/noscript.css"
    /></noscript>
  </head>
  <body class="is-preload">
    <div id="page-wrapper">
      <!-- Header -->
      <header id="header">
        <h1 id="logo"><a href="index.html">Evolutionary Informatics Lab</a></h1>
        <nav id="nav">
          <ul>
            <li><a href="index.html">Home</a></li>
            <li>
              <a href="research.html">Research</a>
              <ul>
                <li><a href="macro.html">Macroevolution</a></li>
                <li><a href="comp.html">Evolutionary Medicine</a></li>
                <li><a href="phylo.html">Phylogenomics</a></li>
                <li><a href="immuno.html">Immunogenetics</a></li>
              </ul>
            </li>
            <li>
              <a href="current.html">Team</a>
              <ul>
                <li><a href="alex.html">Alex Dornburg</a></li>
                <li><a href="current.html">Current Team Members</a></li>
                <li><a href="alumni.html">Former Team Members</a></li>
              </ul>
            </li>
            <li><a href="publications.html">Publications</a></li>
            <li><a href="software.html">Software</a></li>
            <li><a href="media.html">Media</a></li>
            <li><a href="contact.html" class="button primary">Join Us</a></li>
          </ul>
        </nav>
      </header>

      <!-- Main -->
      <div id="main" class="wrapper style1">
        <div class="container">
          <header class="minor">
            <h3>TOAST | Concatenated Alignments </h3>
          </header>
          <!-- Content -->


     <div class="image fit">
            <img class="toast-bg"
            src="images/TOAST/TOASTpage_alt.jpg" alt=""/>
        <div class="transbox">
            <h3>What is in this section</h3>
            <p>
              In this section we will make concatenated alignments for IQTREE2 or other phylogenetic software and look at a rudimentary way to assess missing data patterns before diving deeper in the next section.
              <br />
              <br />
              <strong>This section assumes you have a folder full of single gene/locus alignments</strong>
              <br />
              <br />
              Note that TOAST can batch align your gene files for you using 
              <a href="https://mafft.cbrc.jp/alignment/software/">MAFFT</a> Please go back to the <a href="toastBUSCO.html">previous page</a> if you need to see this function <br />
              <br />
              <br />
            </p>
            </div>
        </div>
          </div>
        <section id="MissingData">
          <div class="container">
            <div class="section-header">
              <h3>Making a Missing Data Matrix</h3>
Missing data is a common feature of large sequence datasets. Everything from changes in coverage depth, DNA/RNA quality, probe design, assembly method, etc can result in uneven coverage of loci between species.TOAST provides a suite of functions to both explore missing data and reassemble alignments based on user determined thresholds of acceptable representation.     
<br>
We will use the missing_data.tsv file distributed with TOAST in the examples in this section. 
<br>
<br>
Note that you can use the following code to generate a .csv or .tsv file (depending on your preference) of missing data patterns from ANY set of FASTA files in a directory with the following function
<pre style="text-align: left; white-space: pre-line">
<code>setwd("~/Documents/TOAST/concatenatedAnalysis/")
missing<-MissingDataTable(aligned_dir = "~/Documents/TOAST/AlignedLoci/")
write.csv(missing, file="missing_data.csv")</code></pre>
<br />
<br />
<h3>Visualizing General Missing Data Patterns</h3>
Part of the orthology assembly steps in both sections 2 and 3 was used to generate .csv file entitled "missing_data.csv" that is <a href='https://github.com/carolinafishes/toast/tree/master/sample_data'> distributed with this software.</a> We will use this file throughout this section. Let's begin by reading it into memory. Note you may need to change the path depending on where you downloaded this. 
<pre style="text-align: left; white-space: pre-line">
<code>missing<-read.csv("/toast/sample_data/missing_data.csv", header=TRUE)</code></pre>
With this file we can begin to explore coarse missing data patterns. 
<br />
If you look at the distributed file, you have probably realized that ANY delimited file of data presence/absence can be used with the following visualization functions (just needs data and NA for no data). This includes phenotypic trait data, behavioural observations, etc!
<br>
<br>
<strong style= color:red> Now you have no excuse to not look at missing data patterns anymore! </strong>
<br />
<br />

<br> note that this data has the loci as a column. Let's quickly remove these. Your data may not need this, but this pops up often so why not cover it here. 
 <img class="image fit" src="images/TOAST/TOAST_headmissing.png" style="width: 75%; height: 75%" alt="">
<pre style="text-align: left; white-space: pre-line">
<code>missing2<-missing[,-1]</code></pre>
<img class="image fit" src="images/TOAST/TOAST_headmissing2.png" style="width: 75%; height: 75%" alt="">
Now we can start by visualizing a missing data matrix for this dataset using the mismap function from the R package Amelia  
<pre style="text-align: left; white-space: pre-line">
<code>library(Amelia)
  par(mar = c(10, 2, 4, 4) + 0.1, oma= c(7, 1, 1, .5) + 0.1)
  missmap(missing2, rank.order=FALSE, las=2, cex.axis=0.8)</code></pre>
Note that later we can use TOAST functions to order this in the order we would like based on a tree, but for now we will use the default<br />
<p> 
  <img class="image fit" src="images/TOAST/TOASTmissmap.png" style="width: 50%; height: 50%" alt="">
  The above gives us a snapshot of how missing data are distributed

<h3>Concatenated Alignments</h3>
It is very easy to generate a concatenated alignment along with a corresponding file that provides the locations of each locus in the alignment<br />
These files are generated by toast for direct input into IQTREE2. The partitions file is in nexus format and the alignment in phylip. Note that TOAST has utilities for converting between filetypes that are covered later in this tutorial.  
<br><br><p>To generate a concatenated alignment you need the directory where your single locus alignments are stored, and the missing object we created above. As you will see in a moment, TOAST can also assemble based on missing data thresholds. </p>
To begin, simply run these two lines of code to generate the files.
  <pre style="text-align: left; white-space: pre-line"><code>SuperAlign("path_to_aligned_directory", missing_df = missing)
    PartitionTable("path_to_aligned_directory", missing_df = missing)
  </code></pre>
This will produce two files, a concatenated alignment called superAlign (This silly name is to encourage you to change the name to something sensible),and "partitionTable". <br>
<br>
<h3>Concatenated Alignments with a Missing Data Threshold</h3>
In case you wish to generate alignments based on a criteria of no more than 100 or no more than 400 missing loci per taxa... you can! 
Simply create a new directory on your laptop where you wan to store the individual fasta files that meet the criteria, and then follow the code below

<br>
<br>You may also be interested in simply knowing which taxa have complete coverage of all orthologs.  
<pre style="text-align: left; white-space: pre-line"><code>#create a second set of fasta files with taxa that have less than 400 missing loci 
#You need to make a new file on your HD called "Fasta_400_missing"
#Set the path here
tff<-"Yourpath/goes/here/Fasta_400_missing"
#Change this number for different thresholds
ThresholdDataTable(missing, threshold=400)->threshold_df
ThresholdExtract(aligned_dir=ad, missing_df=threshold_df, threshold_fasta_folder=tff)</code></pre>
This first step primes a new directory with the files you need. This step is done intentionally for those wishing to only use these files for gene-tree species-tree analyses downstream. From here assemble as you just did. before. 
<pre style="text-align: left; white-space: pre-line"><code>
setwd(tff)
missing<-MissingDataTable(aligned_dir = tff)
write.csv(missing, file="missing_data.csv")
SuperAlign(aligned_dir=tff, missing)
PartitionTable(aligned_dir=tff, missing)
</code></pre>
That's it! In the next section we will do some more advanced visualizations of missing data, or feel free to skip ahead to other utilities such as gene tree based filtration or utility functions. 



<br/>
<br/>
				<a href="toastMain.html" class="button fit ">Next Section: Missing Data Visualization</a>
        <br/>
        <br/>
        <a href="toastBUSCO.html" class="button fit ">Skip to: Gene tree based filtration</a>
        <br/>
        <br/>
        <a href="toastBUSCO.html" class="button fit ">Skip to: Utilities | Interactive Plots</a>
        <br/>
        <br/>
        <a href="toastInstall.html" class="button fit ">Back to: Installation</a>
        <br/>
        <br/>
				<a href="toastMain.html" class="button fit ">Back to: TOAST main page</a>
        <hr />




            
          </div>
        </section>
      

      <!-- Footer -->
      <footer id="footer">
        <ul class="icons">
          <li>
            <a href="#" class="icon brands alt fa-github"
              ><span class="label">GitHub</span></a
            >
          </li>
          <li>
            <a href="#" class="icon solid alt fa-envelope"
              ><span class="label">Email</span></a
            >
          </li>
        </ul>
        <ul class="copyright">
          <li>&copy; Evolutionary Informatics Lab. All rights reserved.</li>
          <li>Design: <a href="http://html5up.net">HTML5 UP</a></li>
        </ul>
      </footer>
    </div>

    <!-- Scripts -->
    <script src="assets/js/jquery.min.js"></script>
    <script src="assets/js/jquery.scrolly.min.js"></script>
    <script src="assets/js/jquery.dropotron.min.js"></script>
    <script src="assets/js/jquery.scrollex.min.js"></script>
    <script src="assets/js/browser.min.js"></script>
    <script src="assets/js/breakpoints.min.js"></script>
    <script src="assets/js/util.js"></script>
    <script src="assets/js/main.js"></script>
  </body>
</html>
