---
layout: page
title: "Considering phylogenetic information in legacy data"
meta_teaser: "How confident are we in past inferences"

header:
    image_fullwidth: "history_banner.jpg"
    caption: Whale breach in Antarctica
    title: ""
image:
    thumb:  history_banner.jpg
    homepage: history_banner.jpg
categories:
    - highlights
comments: true
show_meta: false
---
 
<br> While cruising around the Antarctic I assisted Yale Ph.D. student <a href='https://eeb.yale.edu/people/graduate-students/chantal-parker'>Elyse Parker</a> on analyzing the results of her M.S. work. The resulting paper was <a href="http://carolinafishes.github.io/images/Parker_et_al_2019.pdf"> <en>recently published</en></a> and concerns the need to hold inferences based on legacy markers to the same level of scrutiny as next-gen datasets. I thought I would revist the topic here. 

<h3>Opening cold cases to end incongruence</h3>
<br>
As we move further into the genomics era, a repeating research theme has been to contrast inferences based on hundreds or thousands of markers to historical inferences based on a handful of legacy markers. This contrast is of course necessary to reveal both conflict and congruence in relationships. 
<br>
<br>However, sources of conflict in legacy data are rarely scrutinized while newer datasets are often subjected to a high level of scrutiny through increasingly sophisticated methods. 
<br>
<br>Our manuscript points out that there is a missed opportunity here. Why not <i> also </i> scrutinize available legacy data to see what relationships we have confidence in?
<br>
<br>Using a series of markers commonly used in fish phylogenetics we did just that for the radiation of goodeines, a really cool group of freshwater fishes. 
<br>
<br>We find that there simply is not much information in the markers used for resolving the recent rapid divergences of this clade. 
<br>
<br>
<img class="b30" src="http://carolinafishes.github.io/images/Parkerteal_1.png" alt=""><em> What do we know about goodeine relationships? About this much. We know the major clades but legacy data has little justifiable evidence strongly supporting their inter-relationships.</em>
<br>
<br>
Looking closey within each marker reveals codon positions that either lack of variable sites and thereby contribute little to inference, or high levels of GC3 bias. Scrutinizing the same markers for lack of resolution and uncertainty in triggerfish and squirrelfish relationships revealed the same pathologies. 
<br>
<br>
<img class="b30" src="http://carolinafishes.github.io/images/Parkeretal_2.png" alt=""><em> FIgure from our manuscript summarizing the extant of phylogenetic information in these markers.</em>
<br>
<br> These results are not terribly suprising. The percieved utility of legacy markers was not based on experimental design, but largely on their ability to be readily sequenced and thereby supply data for investigators. Nearly 20 years ago Nick Goldman published a paper referring to this practice as selecting markers based on "folklore". 
<br>
<br> At the time, this was the best we had as phylogeneticists. But now as we move forward towards resolving the Tree of Life, adding assessments of legacy data is a trivial (by computational comparison) way to assess whether a historical conflicting topology is based on strong evidence, noise, or bias. 
<br>
<br> This is particularly important for legacy datasets spanning deeper timescales. 
<br>
<br>
<img class="b30" src="http://carolinafishes.github.io/images/Phillipsetal_1.png" alt=""><em> Informativeness profiles and heatmaps of the probabiliy of correct resolution for a quartet (QIRP) for legacy markers used in leech phylogenetics.</em>
<br>
<br>
I recently aided Bronwyn Williams in an analysis of leech inter-relationships and found that the "missing link" that has been used to explain the evolution of leeches is a highly derived marine leech. The legacy markers were simply noisy and likely driving erroneous placement of this taxon outside of "true leeches". 
<br>
<br>
I would like to add that this is not a condemnation of historical data at all. In fact, I would argue that huge portions of the Tree of Life that were previously resolved have remained virtually unchanged as we added more data. However, there is an opportunity here to assess where we are and are not confident in our inferences. 




